### PR TITLE
refactor-modular-code-a-little-and-add-unit-test-for-it

### PR DESCRIPTION
**Pull request by qiaozha**
As title, to add unit test for modular generation and fix a bug in model inheritence.

### CODE REVIEW

File: packages/typespec-ts/src/modular/emitModels.ts [hunk 1]
```diff
@@ -94,7 +94,13 @@ export function buildModels(
...
           };
         })
-      });
+      };
+      model.type === "model"
+        ? model.parents?.forEach((p) =>
+            modelInterface.extends.push(getType(p).name)
+          )
+        : undefined;
+      modelsFile.addInterface(modelInterface);
```
Comment (qiaozha) [inline packages/typespec-ts/src/modular/emitModels.ts:103]: Fix model inheritence issue found in #1962

File: packages/typespec-ts/test/unit/apiVersion.spec.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { assert } from "chai";
-import { emitClientFactoryFromCadl } from "./util/emitUtil.js";
-import { assertEqualContent } from "./util/testUtil.js";
+import { emitClientFactoryFromCadl } from "../util/emitUtil.js";
```
Comment (lirenhe) [inline packages/typespec-ts/test/unit/apiVersion.spec.ts:2]: Please create an issue for MQ that we need to replace Cadl with TypeSpec/tsp in our codebase.

File: packages/typespec-ts/test/util/emitUtil.ts [hunk 3]
```diff
@@ -204,3 +211,38 @@ export async function getRLCClientsFromCadl(cadlContent: string) {
   expectDiagnosticEmpty(dpgContext.program.diagnostics);
   return clients;
 }
+
+export async function emitModularModelsFromCadl(cadlContent: string) {
```
Comment (lirenhe) [inline packages/typespec-ts/test/util/emitUtil.ts:215]: Can we avoid using 'cadl' in the new code?

File: packages/typespec-ts/src/modular/emitModels.ts (hunk 1 above)
Comment (MaryGao) [reply packages/typespec-ts/src/modular/emitModels.ts:103]: maybe not in this pr, we could integrate inheritance case in cadl-ranch

**Review by MaryGao — APPROVED** [review]
LGTM

File: packages/typespec-ts/test/unit/apiVersion.spec.ts (hunk 2 above)
Comment (qiaozha) [reply packages/typespec-ts/test/unit/apiVersion.spec.ts]: updated in this PR. no need to create an additional issue.

**Pull request merged**